src/phases/query/CollScanComplexPredicate.yml

```diff
@@ -0,0 +1,521 @@
+  Config:
+    ActorFromTemplate:
+      TemplateName: FindQueryTemplate
+      TemplateParameters:
+        Name: {^Parameter: {Name: Name, Default: unused}}
+        ActivePhase: {^Parameter: {Name: ActivePhase, Default: -1}}
+        Filter: *DNFQuery
+
+- TemplateName: ElemMatchCNFTemplate
+  Config:
+    ActorFromTemplate:
+      TemplateName: FindQueryTemplate
+      TemplateParameters:
+        Name: {^Parameter: {Name: Name, Default: unused}}
+        ActivePhase: {^Parameter: {Name: ActivePhase, Default: -1}}
+        Filter:
+          {e: {$elemMatch: *CNFQuery}}
+
+- TemplateName: ElemMatchDNFTemplate
+  Config:
+    ActorFromTemplate:
+      TemplateName: FindQueryTemplate
+      TemplateParameters:
+        Name: {^Parameter: {Name: Name, Default: unused}}
+        ActivePhase: {^Parameter: {Name: ActivePhase, Default: -1}}
+        Filter:
+          {e: {$elemMatch: *DNFQuery}}
+
+Actors:
+- Name: ClearCollection
+  Type: CrudActor
+  Database: *Database
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: 1
+        Collection: *Collection
+        Operations:
+        - OperationName: drop
+
+- Name: InsertData
+  Type: Loader
+  Threads: 4
+  Phases:
+    OnlyActiveInPhases:
+      Active: [1]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *Database
+        MultipleThreadsPerCollection: true
+        CollectionCount: 1
+        DocumentCount: *DocumentCount
+        BatchSize: 1000
+        Document:
+          a: {^RandomInt: {min: 0, max: 10}}
+          b: {^Inc: {start: 11, step: 2}}
+          c: {^RandomInt: {min: 0, max: 1000}}
+          d: {^RandomInt: {min: 0, max: 1000}}
+          e: 
+            a: {^RandomInt: {min: 0, max: 10}}
+            b: {^Inc: {start: 11, step: 2}}
+            c: {^RandomInt: {min: 0, max: 1000}}
+            d: {^RandomInt: {min: 0, max: 100}}
+            e: {
+              a: {^RandomInt: {min: 0, max: 10}},
+              b: {^Inc: {start: 11, step: 2}},
+              c: {^RandomInt: {min: 0, max: 1000}},
+              d: {^RandomInt: {min: 0, max: 100}},
+              e: {a: {^RandomInt: {min: 0, max: 1000}}, b: {^Inc: {start: 11, step: 2}}},
+              f: [{^RandomInt: {min: 0, max: 10}}, {^RandomInt: {min: 0, max: 10}}, {^RandomInt: {min: 0, max: 10}}],
+              g: {^RandomInt: {min: 0, max: 10}},
+              h: {^RandomInt: {min: 0, max: 1000}},
+              i: {^RandomInt: {min: 0, max: 10000}},
+            }
+            f: [{^RandomInt: {min: 0, max: 10}}, {^RandomInt: {min: 0, max: 10}}, {^RandomInt: {min: 0, max: 10}}]
+            g: {^RandomInt: {min: 0, max: 10}}
+            h: {^RandomInt: {min: 0, max: 1000}}
+            i: {^RandomInt: {min: 0, max: 10000}}
+          f: [{^RandomInt: {min: 0, max: 10}}, {^RandomInt: {min: 0, max: 10}}, {^RandomInt: {min: 0, max: 10}}]
```

Comment:
Could also have one of these fields include an array of nested docs with fields that would be matched on by the query above, just to exercise the speed of path traversals, e.g. [{a: 1, ...}, [[a: 2]] ...]